model.py

```diff
@@ -43,12 +40,38 @@
 
 class LLourney(nn.Module):
     
-    def __init__(self):
+    def __init__(
+        self,
+        image_dim: int = 256,
+        patch_dim: int = 2,
+        context_size: int = 32,
+        vae_model_id_or_path: str = 'runwayml/stable-diffusion-v1-5',
+        llm_model_id_or_path: str = 'gpt2',
+        llm_tokenizer_model_id_or_path: str = 'gpt2',
+    ):
         super().__init__()
+        self.image_dim = image_dim
+        self.patch_dim = patch_dim
+        self.context_size = context_size
+
+        self.vae = AutoencoderKL.from_pretrained(vae_model_id_or_path, subfolder='vae')
+        self.vae_scale_factor = 2 ** (len(self.vae.config.block_out_channels) - 1)
+        self.vae_latent_channels = self.vae.config.latent_channels
```

Comment:
Not major, but I might prefer calling this latent_img_channels or something like it. Doesn't seem intrinsic to the VAE.